core/src/main/scala/com/raphtory/components/querymanager/QueryManager.scala

```diff
@@ -14,32 +14,32 @@ import org.apache.pulsar.client.api.Schema
 import scala.collection.mutable
 
 /** @DoNotDocument */
-class QueryManager(scheduler: Scheduler, conf: Config, pulsarController: PulsarController)
-        extends Component[QueryManagement](conf: Config, pulsarController: PulsarController) {
-  private val currentQueries                            = mutable.Map[String, QueryHandler]()
-  private val watermarkGlobal                           = pulsarController.globalwatermarkPublisher()
-  private val watermarks                                = mutable.Map[Int, WatermarkTime]()
-  var cancelableConsumer: Option[Consumer[Array[Byte]]] = None
+class QueryManager(scheduler: Scheduler, conf: Config, topics: TopicRepository)
+        extends Component[QueryManagement](conf) {
+  private val currentQueries = mutable.Map[String, QueryHandler]()
+  //private val watermarkGlobal                           = pulsarController.globalwatermarkPublisher() TODO: remove?
+  private val watermarks     = mutable.Map[Int, WatermarkTime]()
+
+  private val listener = topics
+    .registerListener(
+            s"$deploymentID-query-manager",
+            handleMessage,
+            Seq(topics.submissions, topics.watermark, topics.endedQueries)
+    )
 
   val globalWatermarkMin  = QueryTelemetry.globalWatermarkMin(deploymentID)
   val globalWatermarkMax  = QueryTelemetry.globalWatermarkMax(deploymentID)
   val totalQueriesSpawned = QueryTelemetry.totalQueriesSpawned(deploymentID)
 
   override def run(): Unit = {
     logger.debug("Starting Query Manager Consumer.")
-
-    cancelableConsumer = Some(pulsarController.startQueryManagerConsumer(messageListener()))
+    listener.start()
   }
 
   override def stop(): Unit = {
-    cancelableConsumer match {
-      case Some(value) =>
-        value.unsubscribe()
-        value.close()
-      case None        =>
-    }
+    listener.close()
     currentQueries.foreach(_._2.stop())
-    watermarkGlobal.close()
+    // watermarkGlobal.close() TODO: remove?
```

Comment:
Lets turn it off for the time being to simplify the akka/pulsar questions and then can turn back on/fix it if people require it.